yarn-project/aztec-node/src/aztec-node/server.ts

```diff
@@ -168,6 +171,27 @@ export class AztecNodeService implements AztecNode, AztecNodeAdmin, Traceable {
       );
     }
 
+    const publicClient = createPublicClient({
+      chain: ethereumChain.chainInfo,
+      transport: fallback(config.l1RpcUrls.map(url => http(url))),
+      pollingInterval: config.viemPollingIntervalMS,
+    });
+
+    const registry = getContract({
+      address: config.l1Contracts.registryAddress.toString(),
+      abi: RegistryAbi,
+      client: publicClient,
+    });
+
+    const version = Number(await registry.read.getVersionFor([config.l1Contracts.rollupAddress.toString()]));
```

Comment:
Atm we are kinda abusing the registry and not using it as intended, so this could cause you some issues. I think it would probably be better if we were to use `getVersion` function directly on the rollup instead. That way we at least know that the rollup and the node agrees and don't have to both with what the registry says